site/src/pages/api/blocks.api.ts

```diff
@@ -1,125 +1,49 @@
-import type { NextApiHandler } from "next/";
-
-/** @todo type as JSON object */
-export type BlockProps = object;
-
-/** @todo make part of blockprotocol package and publish */
-export type BlockVariant = {
-  description?: string;
-  displayName?: string;
-  icon?: string;
-  properties?: BlockProps;
+import { NextApiHandler } from "next";
+import { BlockMetadata } from "../../lib/blocks";
+import { formatErrors } from "../../util/api";
+import blocksData from "../../../blocks-data.json";
+
+const validateApiKey = (apiKey: string | string[] | undefined) => {
+  if (!apiKey || typeof apiKey !== "string") {
+    return false;
+  }
+  // @todo add logic to validate API Key
+  return true;
 };
 
-/** @todo make part of blockprotocol package and publish */
-export type BlockMetadata = {
-  author?: string;
-  description?: string;
-  displayName?: string;
-  externals?: Record<string, string>;
-  icon?: string;
-  image?: string;
-  lastUpdated?: string;
-  license?: string;
-  name: string;
-  schema?: string;
-  source?: string;
-  variants?: BlockVariant[];
-  version?: string;
-
-  // @todo should be redundant to block's package.json#name
-  packagePath: string;
-};
+type Response = { results: BlockMetadata[] } | { errors: any };
 
-export type BuildConfig = {
-  workspace: string;
-  repository: string;
-  branch: string;
-  distDir: string;
-  timestamp: string;
-};
+const blocks: NextApiHandler<Response> = (_req, res) => {
```

Comment:
Since `req` is not longer an unused parameter:
```suggestion
const blocks: NextApiHandler<Response> = (req, res) => {
```

Reply:
Thanks. Fixed here 59570d6